bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/obsComp/TargetObsComp.java

```diff
@@ -73,25 +74,27 @@ public Object clone() {
 
     public String getTitle() {
         // Always compute the name; the UI disallows changes.
-        final TargetEnvironment env = getTargetEnvironment();
-        final SPTarget tp = env.getBase();
-        if (tp != null) {
-            final String initName  = tp.getName();
-            final String finalName = initName == null || initName.trim().isEmpty() ? "<Untitled>" : initName;
-            return helper.targetTag(tp.getTarget()) + ": " + finalName;
-        } else {
-            return super.getTitle();
-        }
+        final Asterism asterism = getAsterism();
+        final String initName  = asterism.name();
+        final String finalName = initName == null || initName.trim().isEmpty() ? "<Untitled>" : initName;
+        return helper.targetTag(asterism) + ": " + finalName;
     }
 
     /**
-     * Get the target environment.
+     * Get the target environment (never null).
      */
     public TargetEnvironment getTargetEnvironment() {
         return targetEnv;
     }
 
     /**
+     * Get the target environment (never null).
+     */
+    public Asterism getAsterism() {
+      return getTargetEnvironment().getAsterism();
+    }
+
```

Comment:
@sraaphorst would you expand on this?  The idea is that by design a target component always has a target environment and a target environment always has an asterism (which in turn always has at least one start).